scripts/wrappers/addons.py

```diff
@@ -211,10 +211,28 @@ def update(name: str):
         click.echo("Error: built-in repository '{}' cannot be updated".format(name), err=True)
         sys.exit(1)
 
-    commit_before_pull = git_current_commit(repo_dir)
-
     click.echo("Updating repository {}".format(name))
-    subprocess.check_call([GIT, "pull"], cwd=repo_dir)
+    remote_url = subprocess.check_output(
+        [GIT, "remote", "get-url", "origin"], cwd=repo_dir, stderr=subprocess.DEVNULL
+    ).decode()
+    if remote_url.startswith(snap()):
+        # This is a repository that we have in the snap.
+        # If the branch name we follow has not changed a simple git pull is enough
+        # If the branch name changed we need to git repo add --force
+        followed_branch_name = subprocess.check_output(
+            [GIT, "rev-parse", "--abbrev-ref", "HEAD"], cwd=repo_dir, stderr=subprocess.DEVNULL
+        ).decode()
```

Comment:
I'd add a `.strip()`, but this is fine as well, we only want to compare branches.